x-pack/plugins/alerting/server/authorization/alerting_authorization.ts

```diff
@@ -344,11 +346,28 @@ export class AlertingAuthorization {
           }
         },
       };
+    } else {
+      if (filterOpts.includeSpaceId) {
```

Comment:
If we have a new `asFiltersBySpaceId` function as suggested below, it could handle `includeSpaceId` true/false, so we don't need this extra if/else block here?

Reply:
Can we also leave some comments somewhere here or in the alerting authorization kuery file as to why we need this special case and where it is used?

Reply:
Definitely agree this isn't great. If it helps I cringed committing it 😅 

And yea, I think you're right, that added function can get rid of this extra else block. And will definitely add comments for context!

Reply:
Updated here https://github.com/elastic/kibana/pull/105333/commits/4f8972a0666b3e7a325939b64b77546540b31b06

Thanks!